Sources/PrimerSDK/Classes/Core/PrimerHeadlessUniversalCheckout/Managers/VaultManager.swift

```diff
@@ -40,24 +42,80 @@ extension PrimerHeadlessUniversalCheckout {
+            if vaultedPaymentMethod.paymentMethodType == "PAYMENT_CARD" {
+                let cardNetwork = CardNetwork(cardNetworkStr: vaultedPaymentMethod.paymentInstrumentData.network ?? "")
+                
+                if let vaultedCardAdditionalData = vaultedPaymentMethodAdditionalData as? PrimerVaultedCardAdditionalData {
+                    if vaultedCardAdditionalData.cvv.isEmpty {
+                        let err = PrimerValidationError.invalidCvv(
+                            message: "CVV cannot be blank.",
+                            userInfo: [
+                                "file": #file,
+                                "class": "\(Self.self)",
+                                "function": #function,
+                                "line": "\(#line)"
+                            ],
+                            diagnosticsId: UUID().uuidString)
+                        errors.append(err)
+    
+                    } else if !vaultedCardAdditionalData.cvv.isValidCVV(cardNetwork: cardNetwork) {
+                        let err = PrimerValidationError.invalidCvv(
+                            message: "CVV is not valid.",
+                            userInfo: [
+                                "file": #file,
+                                "class": "\(Self.self)",
+                                "function": #function,
+                                "line": "\(#line)"
+                            ],
+                            diagnosticsId: UUID().uuidString)
+                        errors.append(err)
+                        
+                    } else {
+                        self.vaultedPaymentMethodAdditionalData = vaultedCardAdditionalData
+                    }
+                    
+                    return errors.isEmpty ? nil : errors
+                } else {
+                    let err = PrimerValidationError.vaultedPaymentMethodAdditionalDataMismatch(
+                        paymentMethodType: vaultedPaymentMethod.paymentMethodType,
+                        validVaultedPaymentMethodAdditionalDataType: String(describing: PrimerVaultedCardAdditionalData.self),
+                        userInfo: [
+                            "file": #file,
+                            "class": "\(Self.self)",
+                            "function": #function,
+                            "line": "\(#line)"
+                        ],
+                        diagnosticsId: UUID().uuidString)
+                    errors.append(err)
+                    return errors
+                }
+            } else {
+                // There's no need to validate additional data for payment methods other than PAYMENT_CARD.
+                // Return nil to continue
+                return nil
+            }
+        }
+        
+        public func validate(vaultedPaymentMethodId: String, vaultedPaymentMethodAdditionalData: PrimerVaultedPaymentMethodAdditionalData, completion: @escaping (_ errors: [Error]?) -> Void) {
```

Comment:
I have returned empty list in case validation is fine. Do you think we should align (one way or another)?

Reply:
Either is fine